### PR TITLE
Bugfix/device group lookup

### PR DESCRIPTION
**Pull request by ethanculler**
This should allow groups to be managed appropriately both from the exported resource and as part of the host. I have tested in a local puppet environment, still requires validation from official test suite.

### CODE REVIEW

File: lib/puppet/provider/logicmonitor.rb [hunk 1]
```diff
@@ -197,7 +197,11 @@ def get_device_group(connection, full_path, fields=nil)
                       'device/groups',
                       HTTP_GET,
                       build_query_params("fullPath:#{full_path}", fields, 1))
-
+    debug group_json
+    # debug rest(connection,
```
Comment (samjdacanay) [inline lib/puppet/provider/logicmonitor.rb]: Should we remove the commented lines?

File: lib/puppet/provider/logicmonitor.rb [hunk 2]
```diff
@@ -237,7 +241,7 @@ def recursive_group_create(connection, full_path, description, properties, disab
     debug "Checking for parent device group: #{path[0]}"
     parent_id = 1
     unless nil_or_empty?(parent_path)
-      parent = get_device_group(connection, parent_path, 'id')
+      parent = get_device_group(connection, parent_path.sub('/',''), 'id')
```
Comment (samjdacanay) [inline lib/puppet/provider/logicmonitor.rb]: Should we restrict this to only a leading slash like the other regex?

File: lib/puppet/provider/device_group/device_group.rb [hunk 3]
```diff
@@ -88,7 +88,7 @@ def exists?
     start = Time.now
     debug "Checking if device group \"#{resource[:full_path]}\" exists"
     connection = self.class.get_connection(resource[:account])
-    if resource[:full_path].eql?('/')
+    if resource[:full_path].eql?('/') || resource[:full_path].eql?('')
```
Comment (samjdacanay) [inline lib/puppet/provider/device_group/device_group.rb]: should we check the nil case too?

Comment (ethanculler) [reply lib/puppet/provider/device_group/device_group.rb]: hmm good question... I'll go back through the logic to see if there is a better way to handle this.

File: lib/puppet/provider/logicmonitor.rb (hunk 2 above)
Comment (ethanculler) [reply lib/puppet/provider/logicmonitor.rb]: yes. good catch

Comment (ethanculler) [reply lib/puppet/provider/logicmonitor.rb]: oh wait. this was updated in a later commit.

File: lib/puppet/provider/device_group/device_group.rb (hunk 3 above)
Comment (ethanculler) [reply lib/puppet/provider/device_group/device_group.rb]: It appears you would have to actively set the fullpath to nil in order to cause this to be nil. i think that we should hold off for this PR, and if we need to address we can do so in a follow up PR.

**Pull request merged**